packages/utils/src/modules-sdk/internal-module-service-factory.ts

```diff
@@ -63,6 +63,48 @@ export function internalModuleServiceFactory<
       return keys.map((k) => data[k]).join("_")
     }
 
+    static applyDefaultOrdering(config: FindConfig<any>) {
+      if (config.order) {
+        return
+      }
+
+      const order: object[] = []
+
+      const primaryKeys = AbstractService_.retrievePrimaryKeys(model)
+      const mainEntityOrder = {}
+      primaryKeys.forEach((pk) => {
+        mainEntityOrder[pk] = "ASC"
+      })
+      order.push(mainEntityOrder)
+
+      if (config.relations?.length) {
+        config.relations.forEach((relation) => {
+          const relation_ = (
+            model.__meta ?? model.prototype.__meta
+          ).relations.find((r) => r.name === relation)
+          const relationPrimaryKeys = relation_.targetMeta.primaryKeys ?? ["id"]
+
+          // create order deep object based on relation segment
+          const relationOrder = {}
+          relation.split(".").reduce((acc, segment, index, segments) => {
```

Comment:
This part is really tough to understand. I ran through it with an example and I feel like I'm missing something :D